src/Common/src/TypeSystem/Common/TypeDesc.cs

```diff
@@ -456,5 +456,31 @@ public virtual MethodDesc GetFinalizer()
         {
             return null;
         }
+
+        public bool RequiresAlign8
+        {
+            get
+            {
+                if (Context.Target.Architecture != TargetArchitecture.ARM)
+                {
+                    return false;
+                }
+
+                if (IsArray)
+                {
+                    var elementType = ((ArrayType)this).ElementType;
+                    if ((elementType.IsValueType || elementType.IsEnum) && ((DefType)elementType).InstanceByteAlignment > 4)
```

Comment:
I'll change.  Too used to reading the Binder's code which uses IsValueTypeOrEnum :)